### PR TITLE
Less memory for logging

### PR DESCRIPTION
**Pull request by kbrock**
When we are in log only mode, give option to not record queries.
This will cut down on memory use for queries

https://www.pivotaltracker.com/story/show/140076603

viewing [[diff with whitespace off]](https://github.com/ManageIQ/manageiq-performance/pull/15/files?w=1) may be easier to review

### CODE REVIEW

File: lib/manageiq_performance/middlewares/active_record_queries.rb [hunk 1]
```diff
@@ -130,12 +132,19 @@ def should_measure?
           Thread.current[:miq_perf_sql_query_data]
         end
 
+        def include_queries?
+          return @include_queries if defined?(@include_queries)
+          @include_queries = ManageIQPerformance.config.include_sql_queries?
```
Comment (NickLaMuro) [inline lib/manageiq_performance/middlewares/active_record_queries.rb]: What advantage are we gaining by moving this two a 2 line statement?  You do this on the following two methods as well.

File: lib/manageiq_performance/middlewares/active_record_queries.rb [hunk 2]
```diff
@@ -72,7 +72,7 @@ def call(name, start_time, finish_time, id, payload)
 
               record_sql_query payload[:sql],
                                elapsed_time,
-                               binds_to_params(payload[:binds])
+                               payload[:binds]
```
Comment (NickLaMuro) [inline lib/manageiq_performance/middlewares/active_record_queries.rb]: I like this being moved now to when we use it versus parsing before hand.  👍 

For the record, I was probably under some weird delusion thinking that passing in the full payload was going to cause memory bloat, even though I should know by now that Ruby is "pass by reference".

File: lib/manageiq_performance/middlewares/active_record_queries.rb (hunk 1 above)
Comment (kbrock) [inline lib/manageiq_performance/middlewares/active_record_queries.rb]: if `@include_queries == false`, then `||=` will access `config` every time.

So we can either remove the `@` variable, or we want to cache it whether it is `true` or `false`

Comment (NickLaMuro) [inline lib/manageiq_performance/middlewares/active_record_queries.rb]: Huh, I guess I thought that `||=` was a `.nil?` check and not a "truthy" check.  Works for me.

**Comment by NickLaMuro** [comment]
Going to quickly exercise this option when running with the profiler locally, but I think we can merge this if all goes well.

**Comment by NickLaMuro** [comment]
@kbrock So one thing I noticed with this, and is something that really want your opinion on since I could go either way, but noticed that the "query (ms)" in the `miqperf report --last` was empty when using this setting.


```console
$ miqperf report --last
/vm_cloud/explorer
|     ms | queries | query (ms) |   rows |
|   ---: |    ---: |       ---: |   ---: |
|   8838 |      68 |         41 |   1484 |
|    409 |      66 |         29 |    144 |
|    359 |      66 |         29 |    144 |

$ # v.s with your changes

$ miqperf report --last
/vm_cloud/explorer
|     ms | queries | query (ms) |   rows |
|   ---: |    ---: |       ---: |   ---: |
|   8838 |      68 |          0 |   1484 |
|    409 |      66 |          0 |    144 |
|    359 |      66 |          0 |    144 |

```

This mostly due to how I programmed the middleware storages to communicate via an API versus having them know more about then they needed about the middleware plugins (single responsibility principle of something...), but this comes at a bit of a cost...

### tl; dr;

Do we care that that `query (ms)` value in the report is now going to be blank with this option set, even when the `File` storage is enabled?  (my answer:  we should probably address this later in another PR)

### Cue the long explanation...

So, when writing the middleware storages, I kept things in a "short form" and "long form" to allow the storages to take the data that the needed depending on what was needed for them.  Since there are only two, the `Log` takes the short "summary" form, and the `File` takes the "long" verbose form to do calculations on it later with the `Reporter` class.  Also, by using procs, it avoided any calculations that might be done in them until they were called ("blah blah blah... get to the point").

If you look at the `short` form for both [middlewares/active_record_queries.rb](https://github.com/ManageIQ/manageiq-performance/blob/981972f/lib/manageiq_performance/middlewares/active_record_queries.rb#L42-L43) and [middlewares/active_support_timers.rb](https://github.com/ManageIQ/manageiq-performance/blob/981972f/lib/manageiq_performance/middlewares/active_support_timers.rb#L50-L53), the `active_record_queries` class only keeps track of number of queries and rows, and the time taken by those queries is handled in the `active_support_timers` class.  But if you look in the [reporter.rb](https://github.com/ManageIQ/manageiq-performance/blob/981972f/lib/manageiq_performance/reporter.rb#L124-L126), the value for `queries (ms)` is taken from largest value of either the `active_support_timers` value or the sum of all of the `elapsed_time` for the queries measured from `active_record_queries`.

Right now with your code, the data doesn't exist for the `active_record_queries`, but so the one for `active_support_timers` will be used, and is sometimes `0` even though the multiple queries have been executed during the request.  This has always been the case in the `Log`, since I didn't want to handle "merging" hash keys, and because I was avoiding the calculation of the report data until after the request had finished (in the reporter.rb).  This was probably premature optimization, and most likely the `queries (ms)` could be calculated during the `#record_sql_query` method.

Again, that requires more work than it probably is worth for this change, so I would say "no", but it might be confusing to an end user this bit of conflicting info.  Mostly something to consider in the future.

\#sorryForTheNovel

**Comment by NickLaMuro** [comment]
I am going to make an executive (as not an executive) decision and say let's just merge this.  Wanted to bring up what I found, but it doesn't make sense to fix this here until we have a broader discussion on what should be reporting on the SQL queries (might mean we don't report on the `active_record` value in the `ActiveSupportTimers` in the short form).

Again, that is out of scope for what is being done here, and I suspect we will have more to look at later with this.  Thanks.
